src/test/java/htsjdk/variant/vcf/VCFIteratorTest.java

```diff
@@ -0,0 +1,90 @@
+* copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the
+* Software is furnished to do so, subject to the following
+* conditions:
+* 
+* The above copyright notice and this permission notice shall be
+* included in all copies or substantial portions of the Software.
+* 
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+* OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+* NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+* HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+* WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+* FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
+* THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+package htsjdk.variant.vcf;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.StreamSupport;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import htsjdk.variant.VariantBaseTest;
+
+public class VCFIteratorTest extends VariantBaseTest {
+
+    @DataProvider(name = "VcfFiles")
+    public Object[][] getVcfFiles() {
+        return new Object[][] { 
+                new Object[] { "src/test/resources/htsjdk/tribble/tabix/testTabixIndex.vcf", 25 },
+                new Object[] { "src/test/resources/htsjdk/tribble/tabix/testTabixIndex.vcf.gz", 25 },
+                new Object[] { "src/test/resources/htsjdk/variant/serialization_test.bcf", 12 } };
+    }
+
+    private int countVariants(VCFIterator r) {
+        return (int) StreamSupport.stream(Spliterators.spliteratorUnknownSize(r, Spliterator.ORDERED), true).count();
+    }
+
+    @Test(dataProvider = "VcfFiles")
+    public void testUsingUri(String uri, int nVariants) throws IOException {
+        VCFIterator r = new VCFIteratorBuilder().open(uri);
+        Assert.assertNotNull(r.getFileHeader());
+        Assert.assertEquals(countVariants(r), nVariants);
+        r.close();
+    }
+
+    @Test(dataProvider = "VcfFiles")
+    public void testUsingFile(String uri, int nVariants) throws IOException {
+        VCFIterator r = new VCFIteratorBuilder().open(new File(uri));
+        Assert.assertNotNull(r.getFileHeader());
+        Assert.assertEquals(countVariants(r), nVariants);
+        r.close();
+    }
+
+    @Test(dataProvider = "VcfFiles")
+    public void testUsingStreams(String file, int nVariants) throws IOException {
+        InputStream in = new FileInputStream(file); 
+        VCFIterator r = new VCFIteratorBuilder().open(in);
+        Assert.assertNotNull(r.getFileHeader());
+        Assert.assertEquals(countVariants(r), nVariants);
+        r.close();
+        in.close();
```

Comment:
Can these 3 test be combined into one? Can there be round trip validation of the input file to ensure that what comes in is the same as what comes out?